submariner/vxlan-tunnel.md

```diff
@@ -0,0 +1,34 @@
+# Encryption-less connections between clusters with VXLAN
+
+## Summary
+
+All current cable drivers involve encrypting content, which is great for privacy but involves some overhead.
+On private connections, it would be useful to allow unencrypted connections, for example using IP-on-IP or VXLAN.
+
+## Proposal
+
+This enhancement proposes to use VXLAN for un-encrypted connection between clusters. A new VXLAN interface "vxlan-tunnel"
+will be added to the active gateway node of each cluster and it will act as the VTEP. The traffic which is destined to other
+clusters will be forwarded via this interface to the appropriate clusters.
+
+## Design Details
+
+* A new interface "vxlan-tunnel" will be added.
+* The VXLAN interface will be one-to-many, that is a single interface will be used to connect to all the joined clusters.
+* For the VXLAN VTEP IP, the prefix "241" will be used, followed by the rest of the blocks from the endpoint private IP address. For example,
```

Comment:
We keep adding to the reserved 240.0.0.0/4 class E range of IPv4, I wonder if there's a better solution to that.
we are at risk that such range would eventually used for something and submariner would sink :P

@sridhargaddam Do we know if it would be possible to assign IPv6 addresses to the gateways (we could use LLAs) and use those as destination of IPv4 networks?, I know it's quite bizarre, but, technically it may be possible, as in the end, we only need the destination MAC address for routing, but may be the kernel won't support this.

Reply:
@mangelajo I'll have to check if it works. I've seen some conversations in the past where people completely disable IPv6 in the underlay, so even if it works, for such clusters there could be an issue (though a remote possibility). Its worth spending time on this and trying to understand if it works.

In the meantime, as you know, currently we have a hardcoded 240.0.0.0/8 range. Shall we expose that as an environment variable (even if its hidden to the users) so that we can customize it, if necessary, just to mitigate situations where 240.0.0.0/8 is not allowed for some reason.

Reply:
I think it would be ok to make IPv6 stack a requirement if it worked (i'm not sure). The risk of the 240/4 range is dangerous ':)

An env var could help. but to which range would they migrate? 

We probably should book some time in the future (0.11?) to investigate alternatives to this. But for now 241.x.x.x should be fine I think.

Reply:
Raised https://github.com/submariner-io/submariner/issues/1321 to track.